### DISCUSSION TITLE
Any simple example on how to draw shader on the screen?

### DISCUSSION THREAD

**Discussion by debuxxed** [houmain/gpupad#8] [Q&A] [answered]
I can't seem to figure out.
I've turned auto-evaluation and made simple hello world program with fragment shader:
```glsl
#ifdef GL_ES
precision mediump float;
#endif

void main() {
  gl_FragColor = vec4(1.0);
}
```

**Comment by houmain** [comment, marked as answer]
Hi, thanks for asking!
GPUpad is very flexible but makes the simple things still a little bit complicated. I will think about how this can be simplified!

To execute a fragment shader and see the result, one has to create a few items in the session window (like a target to render to, a shader program and a render call).
For now it is easiest, to start with a sample from the Help menu and edit the shader. The simplest one is the sample "Quad". If you open it, and have a look at the session window, there is an item "Program", which you can double click to open it's vertex and fragment shaders. In the fragment shader you can paste your code.

You should add a "#version 120" in the first line (since you are using [gl_FragColor](https://www.reddit.com/r/opengl/comments/aiswwq/gl_fragcolor_has_been_deprecated_in_450_whats_the/)).
